table/block.h

```diff
@@ -211,36 +221,30 @@ class BlockIter final : public InternalIterator {
       : comparator_(nullptr),
         user_comparator_(nullptr),
         data_(nullptr),
-        restarts_(0),
         num_restarts_(0),
-        current_(0),
         restart_index_(0),
+        restarts_(0),
+        current_(0),
         status_(Status::OK()),
-        prefix_index_(nullptr),
         key_pinned_(false),
+        block_contents_pinned_(false),
         key_includes_seq_(true),
```

Comment:
Why do we still need this?

Reply:
block contents are not pinned when it is not cachable (like in mmap files in nonreadonly db). This is not an index/data specific and is not changed by this refactoring.